src/selections-factory.js

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useState, useSelections } from '@nebula.js/stardust';
+
+export function getCellStyle(selected, cell) {
+  const isSelected = selected[0]?.colIdx === cell.colIdx && !!selected.find((s) => s.qElemNumber === cell.qElemNumber);
+  const isExcluded = selected.length && selected[0].colIdx !== cell.colIdx;
+
+  return isSelected ? { 'background-color': '#009845' } : isExcluded ? { 'background-color': '#e8e8e8' } : {};
+}
+
+export function selectCell(selections, cell) {
+  const { api, selected, setSelected } = selections;
+  const { rowIdx, colIdx, qElemNumber } = cell;
+  let newSelected = [];
+
+  // prevent selection in other column
+  if ((selected.length && selected[0].colIdx !== colIdx) || !api) {
+    return;
+  }
```

Comment:
Is this done in the ui as well? I mean after you clicked in one column, we should not listen to events in the other?
I guess it can be something like click outside confirms, but still think you should need two clicks to start selecting in the next column.

Reply:
No, this function is always called when clicking a cell. I guess we need to handle clicking outside, I just checked with the client and as you say, one click confirms, another starts new selections